features/util.py

```diff
@@ -98,7 +112,26 @@ def build_debs(
 
     :return: A list of file paths to debs created by the build.
     """
-    logging.info("--- Creating: {}".format(SOURCE_PR_TGZ))
+    deb_prefix = "{}-{}-".format(series, repo_state_hash())
+    tools_deb_name = "{}ubuntu-advantage-tools.deb".format(deb_prefix)
+    pro_deb_name = "{}ubuntu-advantage-pro.deb".format(deb_prefix)
+    tools_deb_cache_path = os.path.join(UA_DEB_BUILD_CACHE, tools_deb_name)
+    pro_deb_cache_path = os.path.join(UA_DEB_BUILD_CACHE, pro_deb_name)
+
+    if not os.path.exists(UA_DEB_BUILD_CACHE):
+        os.makedirs(UA_DEB_BUILD_CACHE)
+
+    if os.path.exists(tools_deb_cache_path) and os.path.exists(
+        pro_deb_cache_path
+    ):
+        logging.info(
+            "Using debs in cache: {} and {}".format(
+                tools_deb_cache_path, pro_deb_cache_path
+            )
+        )
+        return [tools_deb_cache_path, pro_deb_cache_path]
+
+    logging.info("Creating: {}".format(SOURCE_PR_TGZ))
```

Comment:
Why are we removing the `---` I think it is useful to differentiate our custom messages from other messages in the tests, like pycloudlib or behave

Reply:
No strong reason, I'll add them back